File: Practice/AllDomains/Tutorials/CrackingTheCodingInterview/SortingBubbleSort.js
/*	author:@shivkrthakur  */
process.stdin.resume();
process.stdin.setEncoding('ascii');

var input_stdin = "";
var input_stdin_array = "";
var input_currentline = 0;

process.stdin.on('data', function (data) {
    input_stdin += data;
});

process.stdin.on('end', function () {
    input_stdin_array = input_stdin.split("\n");
    main();
});

function readLine() {
    return input_stdin_array[input_currentline++];
}

/////////////// ignore above this line ////////////////////

function main() {
    var n = parseInt(readLine());
    a = readLine().split(' ');
    a = a.map(Number);

    var swaps = 0;
    for(var i = 0; i < n; i++)
    {
        for(var j = 0; j < n - 1; j++)
        {
            if(a[j] > a[j + 1]){
                var temp = a[j];
                a[j] = a[j+1];
                a[j+1] = temp;
                swaps++;
            }
        }
    }
    console.log("Array is sorted in %d swaps.", swaps);
    console.log("First Element: %d", a[0]);
    console.log("Last Element: %d", a[n-1]);
}

File: Practice/AllDomains/Tutorials/CrackingTheCodingInterview/SortingBubbleSort.cs
/*	author:@shivkrthakur  */
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
class Solution {
    public static void PrintArray(int[] arr)
    {
        arr.ToList().ForEach(x => {Console.Write($"{x},");});
        Console.WriteLine();
    }
    
    static void Main(String[] args) {
        int n = Convert.ToInt32(Console.ReadLine());
        string[] a_temp = Console.ReadLine().Split(' ');
        int[] a = Array.ConvertAll(a_temp,Int32.Parse);
        //PrintArray(a);
        int outerLength = a.Length, innerLength = a.Length-1, noOfSwaps = 0; 
        for(int i = 0; i < outerLength; i++)
        {
            for(int j = 0; j < innerLength; j++)
            {
                if(a[j] > a[j+1])
                {
                    int temp = a[j];
                    a[j] = a[j+1];
                    a[j+1] = temp;
                    noOfSwaps++;
                    //outerLength = j;
                }
                //PrintArray(a);
            }
        }
        Console.WriteLine($"Array is sorted in {noOfSwaps} swaps.");
        Console.WriteLine($"First Element: {a[0]}");
        Console.WriteLine($"Last Element: {a[a.Length-1]}");
    }
}
